ynr/apps/parties/views.py

```diff
@@ -24,16 +46,83 @@ def get_context_data(self, **kwargs):
                 Election, postextraelection__ballot_paper_id=kwargs["election"]
             )
 
-        candidates = (
-            Membership.objects.filter(
-                post_election__election=election, on_behalf_of=party
-            )
-            .select_related("post_election", "person")
-            .order_by("post_election__post__label")
-        )
+        candidates = party.membership_set.select_related(
+            "post_election", "person"
+        ).order_by("post_election__post__label")
 
         context["party"] = party
         context["election"] = election
         context["candidates"] = candidates
 
         return context
+
+
+class PartyListView(ElectionMixin, TemplateView):
+    template_name = "candidates/party-list.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["parties"] = Party.objects.filter(
+            membership__post_election__election=self.election_data
+        ).distinct()
+        return context
+
+
+class PartyDetailView(ElectionMixin, TemplateView):
+    template_name = "candidates/party.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        party_id = kwargs["organization_id"]
```

Comment:
Should this be `party_id = kwargs["legacy_slug"]` - it looks like we've renamed the URL param in the routing?